src/main/java/core/basesyntax/service/impl/DataConverterImpl.java

```
@@ -0,0 +1,52 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.DataConverter;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+
+    @Override
+    public List<FruitTransaction> convertStringsToTransactionsList(List<String> lines) {
+        if (lines == null) {
+            throw new NullPointerException("Input parameter cannot be null");
+        }
+        List<FruitTransaction> fruitTransactions = new ArrayList<>();
+        for (int i = 1; i < lines.size(); i++) {
+            String currentLine = lines.get(i).trim();
+            if (currentLine.isEmpty()) {
+                continue;
+            }
+            fruitTransactions.add(parseLineToTransaction(currentLine));
+        }
+        return fruitTransactions;
+    }
+
+    private FruitTransaction parseLineToTransaction(String line) {
+        String[] parts = line.split(",");
+        if (parts.length != 3) {
+            throw new RuntimeException("Input line is invalid: has more than 3 parts");
+        }
+        String operationCode = parts[0].trim();
+        String fruitName = parts[1].trim();
+        int quantity;
+        try {
+            quantity = Integer.parseInt(parts[2].trim());
+        } catch (NumberFormatException e) {
+            throw new RuntimeException("Input line is invalid: quantity part is invalid");
```

This violates the checklist item: "Don't ignore exceptions in try-catch blocks" and the correction: "throws informative RuntimeExceptions instead of NPE." The NumberFormatException is caught but rethrown as a generic RuntimeException without including the original exception or contextual detail (line content or value). Wrap the original exception and include the offending token/line in the message.